### PR TITLE
Moved payments dashboard queries from metabase into the warehouse

### LINKED ISSUE #1175: Move payments dashboard queries from metabase into the warehouse
There are a few visualizations on the payments dashboard that are written directly into Metabase:
* [Ride duration](https://dashboards.calitp.org/question/178-median-ride-duration-per-route-only-for-tap-on-off-rides) -- Add a duration field to `payments_rides`
* [Ride distance](https://dashboards.calitp.org/question/276-median-ride-distance-per-route-only-for-tap-on-off-rides) -- During cleaning, add a point `geography` for the lat/lng on a transaction; Add a distance field to `payments_rides`
* [First-time customers](https://dashboards.calitp.org/question/258-number-of-first-time-customers-per-week) -- Add an earliest tap field to the customer table

### PR DESCRIPTION
**Pull request by charlie-costanzo**
# Description

Moved the SQL questions written directly in Metabase (which are powering several visualizations on the payments dashboard) to the warehouse

- [x] [Ride duration](https://dashboards.calitp.org/question/178-median-ride-duration-per-route-only-for-tap-on-off-rides) 
  - Added a `duration` field to `payments_rides`
- [x] [Ride distance](https://dashboards.calitp.org/question/276-median-ride-distance-per-route-only-for-tap-on-off-rides)
  - During cleaning, added a point `geography` for the lat/lng on a transaction to `stg_cleaned_device_transactions` 
  - Added a `distance` field to `payments_rides`
    -  Affected the following columns in `payments_rides` as well
      - Added (*but can remove any that we feel are unneeded*):
        - `on_geography`
        - `off_geography`
        - `earliest_tap`
        - `off_latitude`
        - `off_longitude`
      - Renamed:
        - `latitude` -> `on_latitude`
        - `longitude` -> `on_longitude`
- [x] [First-time customers](https://dashboards.calitp.org/question/258-number-of-first-time-customers-per-week) 
  - Added an `earliest_tap` field to the customer table

Resolves #1175 

## Type of change

- [x] New feature

## How has this been tested?
Local dbt dev environment

### CODE REVIEW

File: warehouse/models/payments_views_staging/stg_cleaned_device_transactions.sql [hunk 1]
```diff
@@ -10,11 +10,14 @@ WITH stg_cleaned_device_transactions AS (
...
+        TRIM(location_id) AS location_id
     FROM {{ ref('stg_enriched_device_transactions') }}
     WHERE calitp_dupe_number = 1
 
 )
 
-SELECT * FROM stg_cleaned_device_transactions
+SELECT
+    *,
+    ST_GEOGPOINT(CAST(longitude AS FLOAT64), CAST(latitude AS FLOAT64)) AS geography
```
Comment (mjumbewu) [inline warehouse/models/payments_views_staging/stg_cleaned_device_transactions.sql:22]: :heart_eyes:

Comment (mjumbewu) [inline warehouse/models/payments_views_staging/stg_cleaned_device_transactions.sql:22]: **suggestion:** If you cast the lng/lat above, you won't need to here.

```suggestion
    ST_GEOGPOINT(longitude, latitude) AS geography
```

**Review by mjumbewu — CHANGES_REQUESTED** [review]
Just a couple suggestions. Otherwise, love it!

**Comment by charlie-costanzo** [comment]
> Just a couple suggestions. Otherwise, love it!

Hey thank you so much for the feedback Mjumbe! I just committed your suggestions, ran them locally, and re-requested review

**Comment by charlie-costanzo** [comment]
Heyo! Wondering what you two think about merging this now that the review requests above have been taken care of @mjumbewu @j-meelah

**Review by atvaccaro — APPROVED** [review]
can you run sqlfluff on this locally since it's disabled in CI/CD?

File: warehouse/models/payments_views/payments_rides.sql [hunk 2]
```diff
@@ -100,84 +100,99 @@ second_transactions AS (
...
-LEFT JOIN gtfs_routes_with_participant AS r
-    ON r.participant_id = m.participant_id
-        AND r.route_id = (CASE WHEN t1.route_id != 'Route Z' THEN t1.route_id ELSE COALESCE(t2.route_id, 'Route Z') END)
-        -- here, can just use t1 because transaction date will be populated
-        -- (don't have to handle unkowns the way we do with route_id)
-        AND r.calitp_extracted_at <= DATETIME(TIMESTAMP(t1.transaction_date_time_utc))
-        AND r.calitp_deleted_at > DATETIME(TIMESTAMP(t1.transaction_date_time_utc))
+    *,
+    DATETIME_DIFF(off_transaction_date_time_pacific, transaction_date_time_pacific, MINUTE) AS duration,
+    ST_DISTANCE(on_geography, off_geography) / 1609.34 AS distance
```
[the remark this answers is not in the document]
Comment (atvaccaro) [reply warehouse/models/payments_views/payments_rides.sql:197]: Oops good catch

[the remark this answers is not in the document]
Comment (charlie-costanzo) [reply warehouse/models/payments_views/payments_rides.sql:197]: cool cool, gonna remove the miles calculation and rename to `distance_meters`

**Pull request merged**